File: apps/ruby/routes/body/multipart-file.rb
require 'primate/route'

Route.post(content_type: "multipart/form-data") do |request|
  multipart = request.body.multipart

  baz = begin
    Integer(multipart.form["baz"].to_s, 10)
  rescue ArgumentError, TypeError
    0
  end
  foo = multipart.form["foo"].to_s
  greeting = multipart.files.find { |f| f.field == "greeting" }
  {
    "baz" => baz,
    "foo" => foo,
    "greeting" => {
      "name"    => greeting&.filename,
      "size"    => greeting&.size,
      "type"    => greeting&.content_type,
      "content" => greeting&.io&.read&.force_encoding("UTF-8"),
    },
  }
end

File: apps/python/routes/body/multipart-file.py
from primate import Route


@Route.post(content_type="multipart/form-data")
def handle_post(request):
    result = request.body.multipart()

    try:
        baz = int(str(result.form.get("baz", "")))
    except (ValueError, TypeError):
        baz = 0

    foo = str(result.form.get("foo", ""))
    greeting = next((f for f in result.files if f.field == "greeting"), None)

    return {
        "baz": baz,
        "foo": foo,
        "greeting": {
            "name": greeting.filename if greeting else None,
            "size": greeting.size if greeting else None,
            "type": greeting.content_type if greeting else None,
            "content": greeting.io.read().decode("utf-8") if greeting else None,
        },
    }
